files/usr/bin/cinnamon-xlet-makepot

```diff
@@ -0,0 +1,282 @@
+        help='Change the variable name gettext is assigned to in your files. The default is _.')
+    parser.add_argument('-o', '--potfile', dest='out_file', default=None,
+        help=('Use this option to specify the location for the generated .pot file. By default '
+              '<uuid>/po/<uuid>.pot is used. This is where translators are expecting to find the pot '
+              'file, so you should only use this option if you want the pot file elsewhere for your own use.'))
+    parser.add_argument('dir', help='the path to the applet/desklet/extension directory')
+
+    args = parser.parse_args()
+
+    dir = os.path.abspath(args.dir)
+    if not os.path.exists(dir):
+        print('%s does not exist' % dir)
+        quit()
+
+    uuid = os.path.basename(dir)
+
+    if os.path.exists(os.path.join(dir, 'files', uuid)):
+        dir = os.path.join(dir, 'files', uuid)
+
+    if args.install:
+        do_install(uuid, dir)
+        quit()
+
+    if args.remove:
+        do_remove(uuid)
+        quit()
+
+    if args.out_file is not None:
+        pot_path = os.path.abspath(args.out_file)
+    else:
+        pot_path = os.path.join(dir, 'po', uuid + '.pot')
+
+    pwd = os.getcwd()
+    os.chdir(dir)
+    if args.js or args.python:
+        try:
+            output = subprocess.check_output(["xgettext", "--version"])
+        except OSError:
+            print("xgettext not found, you may need to install the gettext package")
+            quit()
+
+    os.makedirs(os.path.dirname(pot_path), mode=0o755, exist_ok=True)
+
+    pot_exists = False
+    if args.js:
+        print("Scanning JavaScript files...")
+
+        js_files = []
+        for root, dirs, files in os.walk(dir):
+            rel_root = os.path.relpath(root)
+            js_files += [os.path.join(rel_root, file) for file in files if file.endswith('.js')]
+        if len(js_files) == 0:
+            print('none found')
+        else:
+            print('found %i file(s)\n' % len(js_files))
+            command_args = [
+                'xgettext',
+                '--language=JavaScript',
+                '--keyword=%s' % args.keyword,
+                '--output=%s' % pot_path
+            ]
+            command_args += js_files
+            os.system(' '.join(command_args))
+            pot_exists = True
+
+    if args.python:
+        print("Scanning for python files...")
+
+        py_files = []
+        for root, dirs, files in os.walk(dir):
+            rel_root = os.path.relpath(root)
+            py_files += [os.path.join(rel_root, file) for file in files if file.endswith('.py')]
+        if len(py_files) == 0:
+            print('none found')
+        else:
+            print('found %i file(s)\n' % len(py_files))
+            command_args = [
+                'xgettext',
+                '--language=Python',
+                '--keyword=%s' % args.keyword,
+                '--output=%s' % pot_path
+            ]
+            if pot_exists:
```

Comment:
There is a problem with the implementation of the **pot_exists** variable. This variable will be set to **True** whether strings to translate where found or not while scanning JS files. Running the script for the first time on an xlet **without** strings to translate on its JS files and **with** strings to translate on its Python files will fail due to the use of the `--join-existing` argument in conjunction with a path to a non-existent POT file.

I think that the easiest and *safest* way of doing this is by directly checking the existence of the POT file to decide if the `--join-existing` argument should be added or not.

```python
if os.path.exists(pot_path):
    command_args.append('-j')
```

Reply:
The problem with doing it that way is that if the script is run on an applet that already has a pot file, it will just add to the existing one instead of regenerating it like I would expect. I don't think it matters though. AFAICT none of the xlets we have currently meet that criteria, and the -j option would work if there were.

Reply:
> ...it will just add to the existing one instead of regenerating it like I would expect...

I was about to ask you why would you want to regenerate the POT file, but then I remembered how dumb `xgettext` can be sometimes (it doesn't remove obsolete strings and it doesn't update the line reference from a *msgid* whose source code has been moved). And I suddenly remembered why I remove the POT files when I'm doing POT updates in bulk.

👍